datastream/dataset.py

```diff
@@ -170,7 +212,11 @@ def to_concat(inner_indices):
         return to_concat
 
     @staticmethod
-    def combine(datasets):
+    def combine(datasets: List[Dataset]) -> Dataset:
+        '''
+        Zip multiple datasets together so that all combinations of examples
```

Comment:
Not clear from documentation what is actually the result
Make it clear which combinatorial iterator this is producing
ex.
https://docs.python.org/3/library/itertools.html#itertools.product
https://docs.python.org/3/library/itertools.html#itertools.permutations
https://docs.python.org/3/library/itertools.html#itertools.combinations

Reply:
* How are the indices involved?

Reply:
All combinations is the product. How would you want to formulate this?